test/e2e/autoscaling/cluster_autoscaler_scalability.go

```diff
@@ -220,56 +219,37 @@ var _ = framework.KubeDescribe("Cluster size autoscaler scalability [Slow]", fun
 	})
 
 	It("should scale down underutilized nodes [Feature:ClusterAutoscalerScalability4]", func() {
-		underutilizedReservation := int64(float64(memCapacityMb) * 0.01)
-		fullReservation := int64(float64(memCapacityMb) * 0.8)
-		perNodeReplicas := 10
+		perPodReservation := int(float64(memCapacityMb) * 0.01)
+		// underutilizedNodes are 10% full
+		underutilizedPerNodeReplicas := 10
+		// fullNodes are 70% full
+		fullPerNodeReplicas := 70
 		totalNodes := maxNodes
+		underutilizedRatio := 0.3
 
 		// resize cluster to totalNodes
 		newSizes := map[string]int{
 			anyKey(originalSizes): totalNodes,
 		}
 		setMigSizes(newSizes)
-		framework.ExpectNoError(framework.WaitForClusterSize(f.ClientSet, totalNodes, largeResizeTimeout))
-
-		// annotate all nodes with no-scale-down
-		ScaleDownDisabledKey := "cluster-autoscaler.kubernetes.io/scale-down-disabled"
-
-		nodes, err := f.ClientSet.Core().Nodes().List(metav1.ListOptions{
-			FieldSelector: fields.Set{
-				"spec.unschedulable": "false",
-			}.AsSelector().String(),
-		})
-
-		framework.ExpectNoError(addAnnotation(f, nodes.Items, ScaleDownDisabledKey, "true"))
-
-		// distribute pods (using taints)
-		divider := int(float64(len(nodes.Items)) * 0.7)
 
-		fullNodes := nodes.Items[:divider]
-		underutilizedNodes := nodes.Items[divider:]
+		framework.ExpectNoError(framework.WaitForClusterSize(f.ClientSet, totalNodes, largeResizeTimeout))
 
-		framework.ExpectNoError(makeUnschedulable(f, underutilizedNodes))
+		// distribute pods using replication controllers taking up space that should
+		// be empty after pods are distributed
+		underutilizedNodesNum := int(float64(maxNodes) * underutilizedRatio)
+		fullNodesNum := totalNodes - underutilizedNodesNum
 
-		testId2 := "full"
-		labels2 := map[string]string{"test_id": testId2}
-		cleanup2, err := runReplicatedPodOnEachNodeWithCleanup(f, fullNodes, f.Namespace.Name, 1, "filling-pod", labels2, fullReservation)
-		defer cleanup2()
-		framework.ExpectNoError(err)
+		podDistribution := []podBatch{
+			podBatch{numNodes: fullNodesNum, podsPerNode: fullPerNodeReplicas},
+			podBatch{numNodes: underutilizedNodesNum, podsPerNode: underutilizedPerNodeReplicas}}
 
-		framework.ExpectNoError(makeUnschedulable(f, fullNodes))
-
-		testId := "underutilized"
-		labels := map[string]string{"test_id": testId}
-		cleanup, err := runReplicatedPodOnEachNodeWithCleanup(f, underutilizedNodes, f.Namespace.Name, perNodeReplicas, "underutilizing-pod", labels, underutilizedReservation)
+		cleanup := distributeLoad(f, f.Namespace.Name, "10-70", podDistribution, perPodReservation,
+			memCapacityMb, map[string]string{}, largeScaleUpTimeout)
 		defer cleanup()
-		framework.ExpectNoError(err)
-
-		framework.ExpectNoError(makeSchedulable(f, nodes.Items))
-		framework.ExpectNoError(addAnnotation(f, nodes.Items, ScaleDownDisabledKey, "false"))
 
 		// wait for scale down
-		expectedSize := int(float64(totalNodes) * 0.85)
+		expectedSize := int(float64(totalNodes) * 0.6)
```

Comment:
We have 30% underutilized nodes and here we wait for 40% to be deleted - that seems like it will never happen?